### PR TITLE
MINOR: fix linking errors in javadoc

### PR DESCRIPTION
**Pull request by chia7712**
It seems to me we should block PR for javadoc error since the doc error is too noisy to the build log.

The Javadoc error about inaccessible link is NOT included by this PR.

related to #8291

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: clients/src/main/java/org/apache/kafka/common/ClusterResourceListener.java [hunk 1]
```diff
@@ -26,16 +26,16 @@
  * There will be one invocation of {@link ClusterResourceListener#onUpdate(ClusterResource)} after each metadata response.
  * Note that the cluster id may be null when the Kafka broker version is below 0.10.1.0. If you receive a null cluster id, you can expect it to always be null unless you have a cluster with multiple broker versions which can happen if the cluster is being upgraded while the client is running.
  * <p>
- * {@link org.apache.kafka.clients.producer.ProducerInterceptor} : The {@link ClusterResourceListener#onUpdate(ClusterResource)} method will be invoked after {@link org.apache.kafka.clients.producer.ProducerInterceptor#onSend(ProducerRecord)}
- * but before {@link org.apache.kafka.clients.producer.ProducerInterceptor#onAcknowledgement(RecordMetadata, Exception)} .
+ * {@link org.apache.kafka.clients.producer.ProducerInterceptor} : The {@link ClusterResourceListener#onUpdate(ClusterResource)} method will be invoked after {@link org.apache.kafka.clients.producer.ProducerInterceptor#onSend(org.apache.kafka.clients.producer.ProducerRecord)}
```
Comment (chia7712) [inline clients/src/main/java/org/apache/kafka/common/ClusterResourceListener.java:29]: avoid importing ProducerRecord explicitly

File: clients/src/main/java/org/apache/kafka/common/ClusterResourceListener.java [hunk 2]
```diff
@@ -26,16 +26,16 @@
...
+ * but before {@link org.apache.kafka.clients.producer.ProducerInterceptor#onAcknowledgement(org.apache.kafka.clients.producer.RecordMetadata, Exception)} .
  * <p>
- * {@link org.apache.kafka.clients.consumer.ConsumerInterceptor} : The {@link ClusterResourceListener#onUpdate(ClusterResource)} method will be invoked before {@link org.apache.kafka.clients.consumer.ConsumerInterceptor#onConsume(ConsumerRecords)}
+ * {@link org.apache.kafka.clients.consumer.ConsumerInterceptor} : The {@link ClusterResourceListener#onUpdate(ClusterResource)} method will be invoked before {@link org.apache.kafka.clients.consumer.ConsumerInterceptor#onConsume(org.apache.kafka.clients.consumer.ConsumerRecords)}
```
Comment (chia7712) [inline clients/src/main/java/org/apache/kafka/common/ClusterResourceListener.java:32]: avoid importing ConsumerRecords explicitly

File: clients/src/main/java/org/apache/kafka/clients/admin/AlterConsumerGroupOffsetsOptions.java [hunk 3]
```diff
@@ -18,6 +18,8 @@
 
 import org.apache.kafka.common.annotation.InterfaceStability;
 
+import java.util.Map;
+
 /**
  * Options for the {@link AdminClient#alterConsumerGroupOffsets(String, Map)} call.
```
Comment (kkonstantine) [inline clients/src/main/java/org/apache/kafka/clients/admin/AlterConsumerGroupOffsetsOptions.java]:
```suggestion
 * Options for the {@link AdminClient#alterConsumerGroupOffsets(String, Map, AlterConsumerGroupOffsetsOptions)} call.
```

File: clients/src/main/java/org/apache/kafka/clients/admin/ElectLeadersOptions.java [hunk 4]
```diff
@@ -17,10 +17,13 @@
...
 package org.apache.kafka.clients.admin;
 
+import org.apache.kafka.common.ElectionType;
 import org.apache.kafka.common.annotation.InterfaceStability;
 
+import java.util.Set;
+
 /**
- * Options for {@link Admin#electLeaders(ElectionType, Set, ElectLeadersOptions)}.
+ * Options for {@link Admin#electLeaders(ElectionType, Set, ElectLeadersOptions)} .
```
Comment (kkonstantine) [inline clients/src/main/java/org/apache/kafka/clients/admin/ElectLeadersOptions.java]:
```suggestion
 * Options for {@link Admin#electLeaders(ElectionType, Set, ElectLeadersOptions)}.
```

File: clients/src/main/java/org/apache/kafka/clients/admin/ElectPreferredLeadersOptions.java [hunk 5]
```diff
@@ -17,8 +17,10 @@
 
 package org.apache.kafka.clients.admin;
 
+import org.apache.kafka.common.ElectionType;
 import org.apache.kafka.common.annotation.InterfaceStability;
 import java.util.Collection;
```
Comment (kkonstantine) [inline clients/src/main/java/org/apache/kafka/clients/admin/ElectPreferredLeadersOptions.java:23]:
```suggestion

import java.util.Collection;
```

File: clients/src/main/java/org/apache/kafka/clients/admin/OffsetSpec.java [hunk 6]
```diff
@@ -17,6 +17,8 @@
 package org.apache.kafka.clients.admin;
 
 
+import java.util.Map;
+
```
Comment (kkonstantine) [inline clients/src/main/java/org/apache/kafka/clients/admin/OffsetSpec.java]:
```suggestion
import java.util.Map;

```

File: clients/src/main/java/org/apache/kafka/clients/consumer/Consumer.java [hunk 7]
```diff
@@ -249,7 +249,7 @@
     ConsumerGroupMetadata groupMetadata();
 
     /**
-     * @see KafkaConsumer#enforceRebalance)
+     * @see KafkaConsumer#enforceRebalance()
```
Comment (kkonstantine) [inline clients/src/main/java/org/apache/kafka/clients/consumer/Consumer.java]: This has been fixed by: https://github.com/apache/kafka/pull/8291 
Please omit and rebase to get the latest changes.

File: clients/src/main/java/org/apache/kafka/common/ElectionType.java [hunk 8]
```diff
@@ -18,10 +18,11 @@
 package org.apache.kafka.common;
 
 import java.util.Arrays;
+import java.util.Set;
```
Comment (kkonstantine) [inline clients/src/main/java/org/apache/kafka/common/ElectionType.java:23]: rearranging the imports to match the style might make sense here, given that there are only two import statements.

File: clients/src/main/java/org/apache/kafka/common/ElectionType.java [hunk 9]
```diff
@@ -18,10 +18,11 @@
...
 import org.apache.kafka.common.annotation.InterfaceStability;
```
Comment (kkonstantine) [inline clients/src/main/java/org/apache/kafka/common/ElectionType.java]:
```suggestion
import org.apache.kafka.common.annotation.InterfaceStability;

import java.util.Arrays;
import java.util.Set;
```

File: connect/api/src/main/java/org/apache/kafka/connect/source/SourceTask.java [hunk 10]
```diff
@@ -93,8 +93,8 @@ public void commit() throws InterruptedException {
      * also called when a record is filtered by a transformation, and thus will never be ACK'd by a broker.
      * </p>
      * <p>
-     * This is an alias for {@link commitRecord(SourceRecord, RecordMetadata)} for backwards compatibility. The default
-     * implementation of {@link commitRecord(SourceRecord, RecordMetadata)} just calls this method. It is not necessary
+     * This is an alias for {@link #commitRecord(SourceRecord, RecordMetadata)} for backwards compatibility. The default
```
Comment (kkonstantine) [inline connect/api/src/main/java/org/apache/kafka/connect/source/SourceTask.java]: This has been fixed by: https://github.com/apache/kafka/pull/8291 
Please omit and rebase to get the latest changes.

File: connect/api/src/main/java/org/apache/kafka/connect/source/SourceTask.java [hunk 11]
```diff
@@ -105,7 +105,7 @@ public void commit() throws InterruptedException {
      *
      * @param record {@link SourceRecord} that was successfully sent via the producer or filtered by a transformation
      * @throws InterruptedException
-     * @see commitRecord(SourceRecord, RecordMetadata)
+     * @see #commitRecord(SourceRecord, RecordMetadata)
```
Comment (kkonstantine) [inline connect/api/src/main/java/org/apache/kafka/connect/source/SourceTask.java]: This has been fixed by: https://github.com/apache/kafka/pull/8291 
Please omit and rebase to get the latest changes. 

But see below on line 126 for a misformatting that needs to be fixed:
`@{link commitRecord(SourceRecord)}` needs to be `{@link #commitRecord(SourceRecord)}`

File: streams/src/main/java/org/apache/kafka/streams/StreamsMetrics.java [hunk 12]
```diff
@@ -58,9 +57,9 @@
      * @param recordingLevel     the recording level (e.g., INFO or DEBUG) for this sensor.
      * @param tags               additional tags of the sensor
      * @return The added sensor.
-     * @see #addRateTotalSensor(String, String, String, RecordingLevel, String...)
+     * @see #addRateTotalSensor(String, String, String, Sensor.RecordingLevel, String...)
```
Comment (kkonstantine) [inline streams/src/main/java/org/apache/kafka/streams/StreamsMetrics.java]: This has been fixed by: https://github.com/apache/kafka/pull/8314
Please omit and rebase to get the latest changes.

File: streams/src/main/java/org/apache/kafka/streams/StreamsMetrics.java [hunk 13]
```diff
@@ -58,9 +57,9 @@
...
      * @see #removeSensor(Sensor)
-     * @see #addSensor(String, RecordingLevel, Sensor...)
+     * @see #addSensor(String, Sensor.RecordingLevel, Sensor...)
```
Comment (kkonstantine) [inline streams/src/main/java/org/apache/kafka/streams/StreamsMetrics.java]: This has been fixed by: https://github.com/apache/kafka/pull/8314
Please omit and rebase to get the latest changes. 

Same for the rest of the issues in this file probably

**Review by kkonstantine — COMMENTED** [review]
Thanks @chia7712
This looks pretty good overall. I've added suggestions in a few places that needed improvement. 
Also, a few of the issues have been fixed in other PRs, so it'd be good to rebase.

File: clients/src/main/java/org/apache/kafka/common/ElectionType.java (hunk 9 above)
Comment (chia7712) [reply clients/src/main/java/org/apache/kafka/common/ElectionType.java]: Is there a rule of imports layout? I observe the following layout in project.
```java
import org.apache.kafka.common.KafkaFuture;
import org.apache.kafka.common.Node;
import org.apache.kafka.common.acl.AclOperation;
import org.apache.kafka.common.annotation.InterfaceStability;

import java.util.Collection;
import java.util.Set;
```

```
import java.util.Set;
import java.util.concurrent.ConcurrentMap;
import java.util.concurrent.atomic.AtomicInteger;
import org.apache.kafka.clients.ApiVersions;
import org.apache.kafka.clients.producer.Callback;
import org.apache.kafka.common.utils.ProducerIdAndEpoch;
import org.apache.kafka.common.Cluster;
```

**Comment by chia7712** [comment]
> This looks pretty good overall. I've added suggestions in a few places that needed improvement.
Also, a few of the issues have been fixed in other PRs, so it'd be good to rebase.

@kkonstantine thanks for reviews!!!

**Comment by kkonstantine** [comment]
Unfortunately, the tests 
`org.apache.kafka.streams.integration.StandbyTaskEOSIntegrationTest.surviveWithOneTaskAsStandby` 
and `org.apache.kafka.streams.integration.StandbyTaskEOSIntegrationTest.surviveWithOneTaskAsStandby`
failed dependably in both builders. 

But since these were the only failures, and these tests are not related to the changes, I'll go ahead and merge this javadoc improvements to `trunk`.

**Comment by kkonstantine** [comment]
Merged to `trunk`. Thanks for the fixes @chia7712 !

**Pull request merged**